pkg/endpoint/bpf.go

```diff
@@ -1071,6 +1073,19 @@ func (e *Endpoint) SkipStateClean() {
 	e.unlock()
 }
 
+func (e *Endpoint) updatePolicyMapPressureMetric() {
+	if !option.Config.MetricsConfig.BPFMapPressure {
+		return
+	}
+
+	if e.policyMapPressureGauge == nil {
+		e.policyMapPressureGauge = metrics.NewBPFMapPressureGauge(e.policyMap.NonPrefixedName(), policymap.PressureMetricThreshold)
+	}
```

Comment:
This is something that can potentially be in the endpoint constructor.

Reply:
Yes. I wasn't sure if there was a single endpoint constructor or many. Let me try to track them down, please check after.

Reply:
I moved this to be constructed with the policyMap which is deferred from the endpoint constructor as well but at least we are not checking for it on each metric value set.